### PR TITLE
Fix the Oppia Terms of Service website url to show as a link and clickable

### LINKED ISSUE #4760: In Terms of Service the website should be displayed as link and should be clickable
**Describe the bug**
In Terms of Service the website should be displayed as link and should be clickable

**To Reproduce**
1. Go to Menu - Help
2. Click on Terms of Service
3. In the 1st paragraph look at the website link 

**Expected behavior**
The link https://www.oppia.org should be displayed as a link and it should be clickable.
The sentence ' Thanks for your interest in Oppia Foundation Inc., a nonprofit that operates the website located at https://www.oppia.org (the Website).'

Website is mentioned twice in the statement.

**Demonstration**
If applicable, add screenshots or videos to help explain your problem.
Screenshot_20221124-143700

**Environment**
Device/emulator being used: Infinix smart 5A
Android or SDK version (e.g. Android 5 or SDK 21): Android version 11
App version (you can get this through system app settings or via the admin controls menu in-app): 0.10-beta-5e64fae55e

### PR DESCRIPTION
**Pull request by Uticodes**
## Explanation
Fix #4760: In Terms of Service the website should be displayed as link and should be clickable

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-A11y-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

| Screenshot Before |  Screenshot After
|:-------------------------:|:-------------------------:|
| ts| oppia_terms_of_service_click

### CODE REVIEW

**Comment by BenHenning** [comment]
@Uticodes it seems like this PR has additional extra changes. Please make sure to sync it with the latest develop changes (and make sure that your fork also has the latest develop changes).

**Comment by Uticodes** [comment]
@BenHenning PR now in sync with develop, kindly check again.

**Comment by BenHenning** [comment]
Thanks @Uticodes! As discussed in #4760 we ideally want a solution here that's a bit more future proof than just changing the XML (since we plan to generate it long-term). One suggestion in the issue thread was to try and use [LinkifyCompat](https://developer.android.com/reference/kotlin/androidx/core/text/util/LinkifyCompat) for this purpose. Could you please try that and see if it linkifies the URL without needing to modify the XML directly?

File: app/src/main/java/org/oppia/android/app/policies/PoliciesFragmentPresenter.kt [hunk 1]
```diff
@@ -78,6 +80,9 @@ class PoliciesFragmentPresenter @Inject constructor(
       supportsLinks = true,
       supportsConceptCards = false
     )
+
+    val websitePattern = Pattern.compile("https://www.oppia.org")
```
Comment (adhiamboperes) [inline app/src/main/java/org/oppia/android/app/policies/PoliciesFragmentPresenter.kt]: This pattern will only be useful for finding links to this address, but if we would like to find **any** link in the span, we may use something like:
```suggestion
    val websitePattern = Pattern.compile("(http|ftp|https)://([\\w_-]+(?:(?:\\.[\\w_-]+)+))([\\w.,@?^=%&:/~+#-]*[\\w@?^=%&/~+#-])")
```

**Review by adhiamboperes — CHANGES_REQUESTED** [review]
Thanks @Uticodes.

Please also update the video in your description to test the ToS page as opposed to the Privacy Policy page, as well as an explanation of the fix.

**Comment by Uticodes** [comment]
> @Uticodes it seems like this PR has additional extra changes. Please make sure to sync it with the latest develop changes (and make sure that your fork also has the latest develop changes).

Sure will do, thanks @BenHenning

File: app/src/main/java/org/oppia/android/app/policies/PoliciesFragmentPresenter.kt (hunk 1 above)
Comment (Uticodes) [reply app/src/main/java/org/oppia/android/app/policies/PoliciesFragmentPresenter.kt]: cool, let me test this and see if it works, thanks @adhiamboperes

**Comment by Uticodes** [comment]
> Thanks @Uticodes.
> 
> Please also update the video in your description to test the ToS page as opposed to the Privacy Policy page, as well as an explanation of the fix.

sure, on it.

**Comment by Uticodes** [comment]
@BenHenning @adhiamboperes PTAL
I have update branch, PR and the video to show how it works, kindly review again 🙏

**Review by adhiamboperes — COMMENTED** [review]
I think this fix works though I think we should try to have an automated test for this.
Can you use one of the other URL tests as a template?

File: app/src/main/res/values/terms_of_service.xml [hunk 2]
```diff
@@ -4,8 +4,8 @@
 
     <p>
       Hello! Thanks for your interest in Oppia Foundation Inc., a nonprofit that
-      operates the website located at https://www.oppia.org (the "Website"). The
-      following Terms of Use (also referred to as the "Terms") are a contract
+      operates the website located at https://www.oppia.org (the "Website").
+      The following Terms of Use (also referred to as the "Terms") are a contract
```
Comment (BenHenning) [inline app/src/main/res/values/terms_of_service.xml:8]: Is this wrapping actually needed to fix the issue?

File: app/src/main/java/org/oppia/android/app/policies/PoliciesFragmentPresenter.kt [hunk 3]
```diff
@@ -78,6 +80,12 @@ class PoliciesFragmentPresenter @Inject constructor(
       supportsLinks = true,
       supportsConceptCards = false
     )
+
+    val websitePattern = Pattern.compile(
+      "(http|ftp|https)://([\\w_-]+(?:(?:\\." +
+        "[\\w_-]+)+))([\\w.,@?^=%&:/~+#-]*[\\w@?^=%&/~+#-])"
+    )
+    LinkifyCompat.addLinks(binding.policyDescriptionTextView, websitePattern, "https://")
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/android/app/policies/PoliciesFragmentPresenter.kt:88]: Is this something we can add generally to our HtmlParser class? E.g. here: https://github.com/oppia/oppia-android/blob/a2182477dcdda96d8eaeb8815d13d96fd4093380/utility/src/main/java/org/oppia/android/util/parser/html/HtmlParser.kt#L110

File: app/src/main/java/org/oppia/android/app/policies/PoliciesFragmentPresenter.kt [hunk 4]
```diff
@@ -78,6 +80,12 @@ class PoliciesFragmentPresenter @Inject constructor(
       supportsLinks = true,
       supportsConceptCards = false
     )
+
+    val websitePattern = Pattern.compile(
+      "(http|ftp|https)://([\\w_-]+(?:(?:\\." +
+        "[\\w_-]+)+))([\\w.,@?^=%&:/~+#-]*[\\w@?^=%&/~+#-])"
+    )
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/android/app/policies/PoliciesFragmentPresenter.kt:87]: As indicated in my link in the comment below, we currently linkify using Linkify.WEB_URLS. Why isn't that pattern working in our case?

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @Uticodes! This is definitely on the right track, I think. I left two comments--PTAL.

**Pull request closed without merging**